apps/meteor/client/views/admin/users/AdminUserForm.tsx

```diff
@@ -127,25 +135,35 @@ const UserForm = ({ userData, onReload, ...props }: AdminUserFormProps) => {
 
 	const handleCreateUser = useMutation({
 		mutationFn: createUserAction,
-		onSuccess: async (data) => {
+		onSuccess: async ({ user: { _id } }) => {
 			dispatchToastMessage({ type: 'success', message: t('User_created_successfully!') });
 			await eventStats({
 				params: [{ eventName: 'updateCounter', settingsId: 'Manual_Entry_User_Count' }],
 			});
-			goToUser(data.user._id);
+			router.navigate(`/admin/users/created/${_id}`);
 			onReload();
 		},
 		onError: (error) => {
 			dispatchToastMessage({ type: 'error', message: error });
 		},
 	});
 
-	const handleSaveUser = useMutableCallback(async (userFormPayload) => {
-		const { avatar, ...userFormData } = userFormPayload;
-		if (userData?._id) {
+	const handleSaveUser = useMutableCallback(async (userFormPayload: userFormProps) => {
+		const { avatar, passwordConfirmation, ...userFormData } = userFormPayload;
+
+		if (!isNewUserPage && userData?._id) {
 			return handleUpdateUser.mutateAsync({ userId: userData?._id, data: userFormData });
 		}
-		return handleCreateUser.mutateAsync(userFormData);
+
+		return handleCreateUser.mutateAsync(
+			{ ...userFormData, fields: '' },
+			{
+				onSuccess: () =>
```

Comment:
Is this overriding or adding to the `onSuccess` clause declare on the `useMutation` call? Regardless, why is this here instead of there?

Reply:
Neither, the onSuccess of mutateAsync will only run when this specific mutate function runs while the onSucces of the useMutation will run after any successfull mutate. In hindsight the onSuccess of the useMutation is a better place to have this invalidation. Will fix in a sec.